internal/kldrest/mongoreceipts.go

```diff
@@ -74,12 +74,29 @@ func (m *mongoReceipts) connect() (err error) {
 }
 
 // AddReceipt processes an individual reply message, and contains all errors
-func (m *mongoReceipts) AddReceipt(receipt *map[string]interface{}) error {
-	if err := m.collection.Insert(*receipt); err != nil {
-		log.Errorf("Insert failed for object '%.v': %s", *receipt, err)
-		return err
+// To account for any transitory failures writing to mongoDB, it retries adding receipt with a backoff
+func (m *mongoReceipts) AddReceipt(requestID string, receipt *map[string]interface{}) (err error) {
+	retryTimeout := 10 * time.Second
```

Comment:
I would suggest 120 secs as the timeout here.
Also, maybe move these to a `const(` section at the top of the file?